documentation/Sphinx/user_guide/api/api.rst

```diff
@@ -7,75 +7,223 @@
+
+The library can be linked to an application with the ``-lvernier`` flag.
+
+CMake Support
+"""""""""""""
+
+Vernier includes support for building applications using CMake.  In
+order to use this, it is necessary to directory where Vernier has been
+installed has added to your ``$CMAKE_PREFIX_PATH`` environment
+variable.
 
 Fortran
 ^^^^^^^
 .. Note: The following function is currently defined manually to avoid errors
          caused by Breathe expecting C++ syntax.
 .. .. cpp:function:: subroutine vernier_mod::vernier_start::vernier_start(hash_out, region_name)
 
-vernier_start
+The Fortran interface is implemented through four subroutines which
+provide the same functionality as the C++ library.  The interface
+subroutines contained in the ``vernier_mod`` Fortran module are:
+
+* **vernier_init**: initialises the Vernier interface.  Accepts an
+  optional MPI communicator.  Defaults to ``MPI_COMM_WORLD`` if not
+  specified.
+* **vernier_start**: starts a timed region.  Arguments are an integer
+  of ``kind=vik`` and a a string representing the name of the region
+* **vernier_stop**: ends an existing timed region.  Argument is an
+  integer of ``kind=vik`` which represents the start of the region
+* **vernier_write**: writes the profile information to a file
 
-vernier_stop
+The library can be linked to an application with the ``-lvernier
+-lvernier_c -lvernier_f`` flags.
 
-vernier_write
+Guidelines For Use
+^^^^^^^^^^^^^^^^^^
 
-Dos and don'ts
-^^^^^^^^^^^^^^
+To minimise the chances of an error when using Vernier, adhere to the
+following guidelines:
 
 **Do**:
 
 * Initialise MPI before profiling.
-* Nest timed regions nicely (no overlap).
+* Nest timed regions nicely
 
-**Don't**:
+**Do not**:
 
 * Use a singular hash (or "handle") for all regions.
-* Have a stop calliper after any ``return`` statements.
+* Have a stop calliper after any ``return`` statements
+* Overlap timed regions
+* Exceed the maximum label length
 
 Examples
 ^^^^^^^^
 .. TODO: Update the names of the Profiler class and "prof" object, and update
          the instructions accordingly.
 
-**C++**:
+The following simple examples show how to add Vernier API calls to C++
+and Fortran programs.
+
+Note that these examples assume that the Vernier libraries have been
+installed in a directory called ``lib64``.  If you are installing on a
+system based on Debian or a system that is not 64 bit, the libraries
+will instead be installed in a directory called ``lib``.
+
+C++
+"""
+
+The following shows how to add Vernier calls to an MPI C++ program:
+
+.. code-block:: cpp
+
+   #include "mpi.h"
+   #include "vernier.h"
+
+   int
+   main(int argc, char *argv[])
+   {
+     MPI_Init(&argc, &argv);
+
+     meto::vernier.init();
+
+     // Start measuring a region
+     auto vernier_handle = meto::vernier.start("Main region");
+
+     // Work functions go here
+
+     // Stop measuring
+     meto::vernier.stop(vernier_handle);
+
+     // Write
+     meto::vernier.write();
+
```

Comment:
There should be a call to ``meto::vernier.finalize();`` before ``MPI_Finalize();``